openassessment/xblock/static/js/src/lms/components/WaitingStepList.jsx

```diff
@@ -1,12 +1,31 @@
 import React from 'react';
 import moment from 'moment';
 import PropTypes from 'prop-types';
-import { Button, DataTable } from '@edx/paragon';
+import { Button, DataTable, Form } from '@edx/paragon';
+import { Search } from '@edx/paragon/icons';
 
 const getReadableTime = (timestamp) => moment(timestamp).fromNow(true);
 
-const WaitingStepList = ({ studentList, refreshData }) => {
-  const studentListWithTimeAgo = studentList.map((item) => ({
+const currentDate = new Date();
+
+// Calculate the date seven days ago
+const sevenDaysAgo = new Date(currentDate);
+sevenDaysAgo.setDate(currentDate.getDate() - 7);
+
+const myObject = {
+  username: 'sofie_20',
+  graded: 1,
+  graded_by: 1,
+  created_at: sevenDaysAgo,
+  staff_grade_status: 'waiting',
+  workflow_status: '',
+};
+
+const WaitingStepList = ({
+  studentList, refreshData, findUsername, selectableLearners,
+}) => {
+  const finalList = studentList.length ? studentList : [myObject];
+  const studentListWithTimeAgo = finalList.map((item) => ({
```

Comment:
what's the function of this list?

Reply:
It calculates how much time have happend for example have this date:  2023-08-04
returns 5 days ago